client/my-sites/earn/components/add-edit-coupon-modal/test/index.jsx

```diff
@@ -0,0 +1,353 @@
+
+		expect( codeTextbox.value ).toEqual( '' );
+		waitFor( () => randomButton.click() );
+		waitFor( () => {
+			const generatedCode = codeTextbox.value;
+			expect( generatedCode ).not.toEqual( '' );
+			expect( generatedCode.length ).toBeGreaterThan( 3 );
+		} );
+	} );
+
+	test( 'should show discount amount input when discount type is amount', async () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		const discountTypeSelect = getDiscountTypeSelect();
+		const amountOption = getAmountOption();
+		const percentageOption = getPercentageOption();
+		const amountInput = getAmountInput();
+
+		expect( amountOption.selected ).toBe( false );
+		expect( percentageOption.selected ).toBe( true );
+		expect( amountInput ).name = 'discount_percentage';
+		expect( amountInput ).type = 'text';
+		expect( amountInput ).value = '';
+
+		await userEvent.selectOptions( discountTypeSelect, amountOption );
+
+		expect( amountOption.selected ).toBe( true );
+		expect( percentageOption.selected ).toBe( false );
+		expect( amountInput ).name = 'discount_value';
+		expect( amountInput ).type = 'number';
+		expect( amountInput ).value = '';
+	} );
+
+	test( 'should enable select dropdown if duration other than default', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal
+				closeDialog={ closeDialog }
+				coupon={ { duration: COUPON_DURATION_3_MONTHS } }
+			/>
+		);
+		expect( screen.getByRole( 'checkbox', { name: 'Duration' } ) ).toBeChecked();
+		expect( screen.getByRole( 'combobox', { name: 'duration selection' } ) ).not.toBeDisabled();
+	} );
+
+	test( 'should show select dropdown as disabled unless toggled', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		const limitDurationToggle = getLimitDurationToggle();
+		const limitDurationSelection = getLimitDurationSelection();
+		expect( limitDurationToggle ).not.toBeChecked();
+		expect( limitDurationSelection ).toBeDisabled();
+
+		act( () => limitDurationToggle.click() );
+
+		expect( limitDurationToggle ).toBeChecked();
+		expect( limitDurationSelection ).not.toBeDisabled();
+	} );
+
+	test( 'should enable emails textarea if email allow list other than default', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		const email_allow_list = [ '*@*.edu', '*@automattic.com' ];
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal
+				closeDialog={ closeDialog }
+				coupon={ { email_allow_list } }
+			/>
+		);
+		const limitCouponsToggle = getLimitCouponsToggle();
+		const limitCouponsTextInput = getLimitCouponsTextInput();
+		expect( limitCouponsToggle ).toBeChecked();
+		expect( limitCouponsTextInput ).not.toBeDisabled();
+		expect( limitCouponsTextInput.value ).toEqual( email_allow_list.join( ', ' ) );
```

Comment:
We should use `toHaveValue()` instead:

 ```suggestion
		expect( limitCouponsTextInput ).toHaveValue( email_allow_list.join( ', ' ) );
```